springbootsample/build.gradle

```diff
@@ -13,7 +13,9 @@ apply plugin: 'io.spring.dependency-management'
 
 group = 'com.microsoft.applicationinsights'
 version = '0.0.1-SNAPSHOT'
-sourceCompatibility = '1.8'
+if (rootDir != projectDir) {
+  apply from: "$rootDir/gradle/common-java.gradle"
+}
```

Comment:
Rather than defining sourceCompatibility and targetCompatibility in each and every module, it is attempting to grab the definition from root project, which already enforces that. If the sample is built without the root project, then it's not going to enforce the source/target compatibility.